astrality/compiler.py

```diff
@@ -159,15 +157,13 @@ def compile_template(
     target.chmod(template_permissions)
 
     if permissions:
-        mode: int
-        if isinstance(permissions, int):
-            mode = permissions
-        elif isinstance(permissions, str):
-            mode = int(permissions, base=8)
-        else:
-            raise MisconfiguredConfigurationFile(
-                f'Tried to write template "{template}" to path "{target}" '
-                f'with unsupported permission type "{permissions}".',
+        result = run_shell(
+            command=f'chmod {permissions} {target}',
+            timeout=0,
+            fallback=None,
+        )
```

Comment:
``result`` is either ``fallback`` if unsucessful, or the standard output of the command, if successful. If the ``chmod`` shell command is successful, it will return an empty string as its output. If we check ``if not result`` it will evaluate as ``True`` even if it was successful. So I think we should replace ``if not result`` with ``if result is None``.

It should be said that ``astrality.utils.run_shell`` includes a trailing newline for shell commands that return nothing, so the code should not break at the moment. But I consider this an implementation detail, and it probably will change at some point in order to clean up Astrality's logging output.